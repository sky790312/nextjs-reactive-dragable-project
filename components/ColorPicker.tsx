import React, { useState } from 'react'
import styled from 'styled-components'

interface Props {
  defaultColor?: string
  shouldShowText?: boolean
  onColorChange: (color: string) => void
}

const ColorPicker: React.FC<Props> = React.memo(
  ({ defaultColor = '#008800', shouldShowText = true, onColorChange }) => {
    const [color, setColor] = useState(defaultColor)

    const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
      const updateColor = e.target.value
      setColor(updateColor)
      onColorChange(updateColor)
    }

    return (
      <>
        <ColorPickerWrapper color={color}>
          <ColorInput value={color} type='color' onChange={onChange} />
        </ColorPickerWrapper>
        {shouldShowText && color}
      </>
    )
  },
)

export default ColorPicker

const ColorPickerWrapper = styled.div<{ color: string }>`
  width: 16px;
  height: 16px;
  align-self: center;
  overflow: hidden;
  background-color: ${props => props.color};
`

const ColorInput = styled.input`
  opacity: 0;
  display: block;
  width: 32px;
  height: 32px;
  border: none;
  cursor: pointer;
`
