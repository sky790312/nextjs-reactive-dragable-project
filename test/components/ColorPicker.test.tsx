import React from 'react'
import { render, withProviders } from '@/test/testUtils'
import ColorPicker from '@/components/ColorPicker'

describe('ColorPicker component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      withProviders(<ColorPicker onColorChange={jest.fn} />),
      {},
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
