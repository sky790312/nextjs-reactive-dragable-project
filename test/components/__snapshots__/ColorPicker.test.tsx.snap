// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorPicker__ColorPickerWrapper-sc-1fchp4r-0 eSLzde"
    color="#008800"
  >
    <input
      class="ColorPicker__ColorInput-sc-1fchp4r-1 gufyRf"
      type="color"
      value="#008800"
    />
  </div>
  #008800
</DocumentFragment>
`;
